mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala

```diff
@@ -191,6 +191,10 @@ class LogisticRegressionSuite extends SparkFunSuite with MLlibTestSparkContext {
 
     val model = lr.fit(dataset)
     assert(model.numClasses === 2)
+    val datasetFeatureSize = dataset.rdd.first() match {
```

Comment:
You shouldn't assume a column ordering.  Instead:
```
dataset.select("features").first().getAs[Vector](0).size
```
or
```
dataset.select("features").first() match {
  case features: Vector => features.size
}
```

(same issue in other test suites)